src/protocols/secure_channel/CASESession.cpp

```diff
@@ -974,140 +1074,143 @@ CHIP_ERROR CASESession::TryResumeSession(SessionResumptionStorage::ConstResumpti
     {
-        std::copy(initiatorRandom.begin(), initiatorRandom.end(), mInitiatorRandom);
-        std::copy(resumptionId.begin(), resumptionId.end(), mResumeResumptionId.begin());
+        std::copy(parsedSigma1.initiatorRandom.begin(), parsedSigma1.initiatorRandom.end(), mInitiatorRandom);
+        std::copy(parsedSigma1.resumptionId.begin(), parsedSigma1.resumptionId.end(), mResumeResumptionId.begin());
 
-        // Send Sigma2Resume message to the initiator
-        MATTER_LOG_METRIC_BEGIN(kMetricDeviceCASESessionSigma2Resume);
-        err = SendSigma2Resume();
-        if (CHIP_NO_ERROR != err)
-        {
-            MATTER_LOG_METRIC_END(kMetricDeviceCASESessionSigma2Resume, err);
-        }
-        SuccessOrExit(err);
-
-        mDelegate->OnSessionEstablishmentStarted();
+        // Next Step is to send Sigma2Resume message to the initiator
+        mNextStep = Step::kSendSigma2Resume;
 
-        // Early returning here, since we have sent Sigma2Resume, and no further processing is needed for the Sigma1 message
+        //  Early returning here, since the next Step is known to be Sigma2Resume, and no further processing is needed for the
+        //  Sigma1 message
         return CHIP_NO_ERROR;
     }
 
+    //  ParseSigma1 ensures that:
+    //  mRemotePubKey.Length() == initiatorPubKey.size() == kP256_PublicKey_Length.
+    memcpy(mRemotePubKey.Bytes(), parsedSigma1.initiatorEphPubKey.data(), mRemotePubKey.Length());
+
     // Attempt to match the initiator's desired destination based on local fabric table.
-    err = FindLocalNodeFromDestinationId(destinationIdentifier, initiatorRandom);
+    err = FindLocalNodeFromDestinationId(parsedSigma1.destinationId, parsedSigma1.initiatorRandom);
     if (err == CHIP_NO_ERROR)
     {
         ChipLogProgress(SecureChannel, "CASE matched destination ID: fabricIndex %u, NodeID 0x" ChipLogFormatX64,
                         static_cast<unsigned>(mFabricIndex), ChipLogValueX64(mLocalNodeId));
 
         // Side-effect of FindLocalNodeFromDestinationId success was that mFabricIndex/mLocalNodeId are now
         // set to the local fabric and associated NodeId that was targeted by the initiator.
+
+        mNextStep = Step::kSendSigma2;
     }
     else
     {
         ChipLogError(SecureChannel, "CASE failed to match destination ID with local fabrics");
-        ChipLogByteSpan(SecureChannel, destinationIdentifier);
-    }
-    SuccessOrExit(err);
+        ChipLogByteSpan(SecureChannel, parsedSigma1.destinationId);
 
-    // ParseSigma1 ensures that:
-    // mRemotePubKey.Length() == initiatorPubKey.size() == kP256_PublicKey_Length.
-    memcpy(mRemotePubKey.Bytes(), initiatorPubKey.data(), mRemotePubKey.Length());
+        // FindLocalNodeFromDestinationId returns CHIP_ERROR_KEY_NOT_FOUND if validation of DestinationID fails, which will trigger
+        // status Report with ProtocolCode = NoSharedTrustRoots.
```

Comment:
Is this always true?

Reply:
- Actually my comment isn't correct, we can get other return values such as `CHIP_ERROR_INCORRECT_STATE`.
- are you suggesting that we add a `ChipLogError` with some more parameters to be added inside `FindLocalNodeFromDestinationId` when `found == false`, or do you want to modify the `SendStatusReport` itself to communicate those?

Reply:
> ChipLogError with some more parameters to be added inside FindLocalNodeFromDestinationId when found == false

That

Reply:
This will require deeper restructuing of `FindLocalNodeFromDestinationId`, Will look into it for a follow-up.